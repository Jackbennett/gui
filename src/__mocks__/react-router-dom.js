import React from 'react';
import { compose, setDisplayName } from 'recompose';
import * as rrd from 'react-router-dom';
// Just render plain div with its children
// eslint-disable-next-line no-import-assign
rrd.BrowserRouter = compose(setDisplayName('BrowserRouter'))(({ children }) => <div>{children}</div>);

module.exports = rrd;
