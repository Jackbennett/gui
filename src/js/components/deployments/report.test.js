import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { createMount } from '@material-ui/core/test-utils';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import DeploymentReport from './report';

const mockStore = configureStore([thunk]);
let dateMock;

describe('DeploymentReport Component', () => {
  let store;
  beforeEach(() => {
    store = mockStore({
      app: {
        features: {
          isEnterprise: false
        }
      },
      devices: {
        byStatus: {
          accepted: { total: 0 }
        }
      },
      deployments: {
        byId: {
          a1: {
            artifact_name: 'test',
            created: '2019-01-01',
            devices: {},
            finished: '2019-01-01',
            stats: {}
          }
        },
        selectedDeployment: 'a1'
      },
      releases: {
        byId: {
          a1: {
            Name: 'a1',
            device_types_compatible: []
          }
        }
      },
      users: {
        organization: {
          plan: 'os'
        }
      }
    });
    const mockDate = new Date('2019-01-01T13:00:00.000Z');
    const _Date = Date;
    global.Date = jest.fn(() => mockDate);
    global.Date.parse = _Date.parse;
    global.Date.now = _Date.now;
    dateMock = jest.spyOn(global, 'Date').mockImplementation(() => mockDate);
  });

  afterEach(() => {
    dateMock.mockRestore();
  });

  it('renders correctly', () => {
    const tree = createMount()(
      <MemoryRouter>
        <Provider store={store}>
          <DeploymentReport deployment={{ id: 'a1' }} type="finished" />
        </Provider>
      </MemoryRouter>
    ).html();
    expect(tree).toMatchSnapshot();
    // due to the rendering of dialogs with the mui testutils, the following does not succeed, the sub views are handled independently though
    // expect(JSON.stringify(tree)).toEqual(expect.not.stringMatching(undefineds));
  });
});
