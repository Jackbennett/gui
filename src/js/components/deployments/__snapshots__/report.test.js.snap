// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeploymentReport Component renders correctly 1`] = `"<div role=\\"presentation\\" class=\\"MuiDialog-root\\" style=\\"position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;\\"><div class=\\"MuiBackdrop-root\\" aria-hidden=\\"true\\" style=\\"opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\"></div><div tabindex=\\"0\\" data-test=\\"sentinelStart\\"></div><div class=\\"MuiDialog-container MuiDialog-scrollPaper\\" style=\\"opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\" role=\\"none presentation\\" tabindex=\\"-1\\"><div class=\\"MuiPaper-root MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthLg MuiDialog-paperFullWidth MuiPaper-elevation24 MuiPaper-rounded\\" role=\\"dialog\\"><div class=\\"MuiDialogTitle-root\\"><h2 class=\\"MuiTypography-root MuiTypography-h6\\">Deployment details</h2></div><div class=\\"MuiDialogContent-root deployment-report\\" style=\\"overflow: hidden;\\"><div class=\\"MuiTabs-root\\"><div class=\\"MuiTabs-scroller MuiTabs-fixed\\" style=\\"overflow: hidden;\\"><div class=\\"MuiTabs-flexContainer\\" role=\\"tablist\\"><button class=\\"MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected\\" tabindex=\\"0\\" type=\\"button\\" role=\\"tab\\" aria-selected=\\"true\\"><span class=\\"MuiTab-wrapper\\">Devices overview</span><span class=\\"MuiTouchRipple-root\\"></span></button><button class=\\"MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary\\" tabindex=\\"0\\" type=\\"button\\" role=\\"tab\\" aria-selected=\\"false\\"><span class=\\"MuiTab-wrapper\\">Deployment details</span><span class=\\"MuiTouchRipple-root\\"></span></button></div><span class=\\"PrivateTabIndicator-root-103 PrivateTabIndicator-colorSecondary-105 hidden\\" style=\\"left: 0px; width: 0px;\\"></span></div></div><div><div class=\\"report-container\\"><div class=\\"deploymentInfo two-columns\\"><div><div style=\\"margin-bottom: -15px;\\"><li class=\\"MuiListItem-root attributes MuiListItem-gutters Mui-disabled\\" disabled=\\"\\"><div class=\\"MuiListItemText-root MuiListItemText-multiline\\"><span class=\\"MuiTypography-root MuiListItemText-primary MuiTypography-body1\\">Release:</span><div class=\\"MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary\\" title=\\"[object Object]\\"><div><span class=\\" \\"><a style=\\"font-weight: 500;\\" href=\\"/releases/test\\">test</a></span> </div></div></div></li></div><div style=\\"margin-bottom: -15px;\\"><li class=\\"MuiListItem-root attributes MuiListItem-gutters Mui-disabled\\" disabled=\\"\\"><div class=\\"MuiListItemText-root MuiListItemText-multiline\\"><span class=\\"MuiTypography-root MuiListItemText-primary MuiTypography-body1\\">Device group:</span><div class=\\"MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary\\"><div><span class=\\" \\"></span> </div></div></div></li></div></div><div style=\\"margin-bottom: -15px;\\"><li class=\\"MuiListItem-root attributes MuiListItem-gutters Mui-disabled\\" disabled=\\"\\"><div class=\\"MuiListItemText-root MuiListItemText-multiline\\"><span class=\\"MuiTypography-root MuiListItemText-primary MuiTypography-body1\\">Status:</span><div class=\\"MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary\\" title=\\"[object Object]\\"><div><span class=\\" \\"><div>Finished </div></span> </div></div></div></li></div></div></div></div></div><div class=\\"MuiDialogActions-root MuiDialogActions-spacing\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">Close</span><span class=\\"MuiTouchRipple-root\\"></span></button></div></div></div><div tabindex=\\"0\\" data-test=\\"sentinelEnd\\"></div></div>"`;
